platforms/core-runtime/daemon-protocol/src/main/java/org/gradle/launcher/daemon/toolchain/DaemonJvmCriteria.java

```diff
@@ -153,5 +155,9 @@ public JavaLanguageVersion probeJavaLanguageVersion(JvmVersionDetector detector)
         public String toString() {
             return String.format("Compatible with Java %s, %s (from %s)", getJavaVersion(), getVendorSpec(), DaemonJvmPropertiesDefaults.DAEMON_JVM_PROPERTIES_FILE);
```

Comment:
Will be great to add the native image capable on the string since this end up being part of the error message and definitely useful to understand why wasn't able to pick certain toolchain. In fact I'll consider adding it only if `true` to avoid displaying unnecessary info when `false`

Reply:
Good point. I'll add it always for now, as it was done that way for toolchains.

But I have been thinking the same: limit the `JavaToolchainSpec.toString` to be focused on the things that matter: language version, vendor if not `any`, ...